crates/ide_completion/src/render.rs

```diff
@@ -91,14 +91,18 @@ pub(crate) fn render_field(
     );
     item.set_relevance(CompletionRelevance {
         type_match: compute_type_match(ctx.completion, ty),
-        exact_name_match: compute_exact_name_match(ctx.completion, &name),
+        exact_name_match: compute_exact_name_match(ctx.completion, name.as_str()),
         ..CompletionRelevance::default()
     });
     item.kind(SymbolKind::Field)
         .detail(ty.display(ctx.db()).to_string())
         .set_documentation(field.docs(ctx.db()))
         .set_deprecated(is_deprecated)
-        .lookup_by(name);
+        .lookup_by(name.as_str());
+    let is_keyword = SyntaxKind::from_keyword(name.as_str()).is_some();
+    if is_keyword && !matches!(name.as_str(), "self" | "crate" | "super" | "Self") {
+        item.insert_text(String::from("r#") + name.as_str());
```

Comment:
```suggestion
        item.insert_text(format!("r#{}", name));
```

It's better to avoid using `+` in rust to build new strings, as that's usually less efficient than possible. I am no sure if we actually can save re-allocation in this particular case, but sticking to `format!` everywhere allows one to not think about this mostly!

Reply:
Sounds good to me! Should we reopen the PR and commit the changes? Or would you like me to open a second PR with the changes?

Reply:
No, fixed in https://github.com/rust-analyzer/rust-analyzer/pull/10167.

Reply:
Perfect